src/AggregateFunctions/Combinators/AggregateFunctionIf.cpp

```diff
@@ -315,7 +315,7 @@ class AggregateFunctionIfNullVariadic final : public AggregateFunctionNullBase<
 
             for (size_t i = row_begin; i < row_end; i++)
             {
-                final_null_flags[i] = filter_null_map[i] || !filter_values[i];
+                final_null_flags[i] = (!!filter_null_map[i]) | !filter_values[i];
```

Comment:
Could you please explain why does this matter?

Reply:
Previous codes can't be optimized to SIMD instructions by compiler. We need to remove branches in loop to remind the compiler to do vectoriazation.

Reply:
@antaljanosbenjamin I'm sorry. After testing, I found what I said was wrong. The changes could not trigger auto-vectorization by compiler. But it could remove branches in loops. 


Before:

![image](https://github.com/user-attachments/assets/90bf627a-812a-4e3d-a526-d22880515257)


After: 
![image](https://github.com/user-attachments/assets/8baaa149-4aa9-4f91-a301-a1d3af190b6d)